### ISSUE TITLE
New Flux node to read tool values.

### ISSUE THREAD

**Issue by LeCloutPanda** [Yellow-Dog-Man/Resonite-Issues#193]
### Is your feature request related to a problem? Please describe.

I seem to have the issue that I can't tell if a tool is either grabbed or properly equipped and I can't seem to figure out how to get it to work correctly.

### Describe the solution you'd like

Adding a node that has a bool output for if a tool is grabbed and not equipped then modifying the current one to properly say if it is equipped or not.

### Describe alternatives you've considered

Checking if the grabble on the tool is triggered but that still doesn't meet my requirements.

**Comment by ProbablePrime** [comment]
Hey @LeCloutPanda , 

We have both "Is Grabbable Grabbed" and "Is Equipped" style nodes for tools. Why are these not suitable?

**Comment by LeCloutPanda** [comment]
I've attempted using those nodes and have found if an object that has Grabbable on it the "Is Grabbable Grabbed" becomes true only when a laser grab happens when the item has a tool tip on it.

What I'm requesting is a node to be able to tell what state a tool is in rather it be free, grabbed and equipped(as if you equip through context menu).

It could also be that "Is Grabbable Grabbed" is bugged and only goes off if you laser grab and it has a tool tip comp.

**Comment by LeCloutPanda** [comment]
I've come to the realisation it could just be to modify the "Is Tool Equipped" to just have a bool that is "Is Held" or "Is Properly Equipped" again just adding a way to differentiate between just holding the tool and actually equipping the tool.

**Comment by shiftyscales** [comment]
As I mentioned on #611:

> How is the box being grabbed? Physically? If so, is it being soft-equipped as opposed to hard then?
> 
> Does the issue also occur when grabbing remotely (with the laser)?
> 
> There is a checkbox on the RawDataTool component if you need it to not be equipped while held physically.
> 
> "Grabbed" and "Equipped" are two different states. It is either being grabbed, or it is equipped. It can't be both.
> 
> Otherwise- if you are wanting a mechanism to distinguish between soft and hard-equipping- then this wouldn't be a bug issue, but an enhancement.

Since it sounds like the underlying issue you are having is covered in the other issue you made #611 - I am going to close this issue out as a duplicate. Per that issue, if you have answers to the above questions that would be helpful to better understand what you underlying problem is exactly, or if existing options are sufficient.

**Comment by LeCloutPanda** [comment]
I don't think it should be closed off—having a node to determine whether a tool is grip-equipped or properly equipped would be helpful. For instance, if I want a tool to have different controls depending on whether it's grip-equipped or properly equipped, an example could be with a line tool. If it's grip-equipped, I can limit its functionality to drawing a simple line. However, if it's properly equipped, I can allow use of the more advanced features.

**Comment by shiftyscales** [comment]
Re-evaluating this- having a "IsSoftEquipped" node that accepts a tool input and outputs a bool could be useful for contexts in which you wish to determine how a tool is equipped. Thank you for your suggestion, @LeCloutPanda.

**Comment by shiftyscales** [comment]
When coupled with the existing IsEquipped node- this should allow all possible states to be tracked.
IsEquipped = True & IsSoftEquipped = True - The tool is soft-equipped.
IsEquipped = True & IsSoftEquipped = False - The tool is hard-equipped.
IsEquipped = False - The tool is not equipped.
